### PR TITLE
ServiceManager package cleanup

### PR DESCRIPTION
**Pull request by pveentjer**
Aligned the ServiceManager with other services in spi.impl package

* Moved ServiceManager into own package
* interface pull out
* explicit unit tests added
* Moved ServiceManagerLogic that was placed in the NodeEngine into the ServiceManager.

### CODE REVIEW

**Comment by pveentjer** [comment]
ping

**Comment by gurbuzali** [comment]
:+1:

File: hazelcast/src/test/java/com/hazelcast/spi/impl/servicemanager/impl/ServiceManagerImplTest.java [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+import com.hazelcast.spi.impl.NodeEngineImpl;
+import com.hazelcast.test.HazelcastParallelClassRunner;
+import com.hazelcast.test.HazelcastTestSupport;
+import com.hazelcast.test.annotation.QuickTest;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+
+import javax.swing.*;
```
Comment (Donnerbart) [inline hazelcast/src/test/java/com/hazelcast/spi/impl/servicemanager/impl/ServiceManagerImplTest.java]: Swing?

**Comment by Donnerbart** [comment]
Beside the one import in the test file: :+1:

**Comment by pveentjer** [comment]
The import makes sense since I check for a guaranteed non existing class.

**Comment by Donnerbart** [comment]
Ah, `JPanel.class`, there it is. A bit hard to find with the wildcard import on GitHub. Okay :)

**Comment by pveentjer** [comment]
Ahh.. stupid wild cards.. intellij... pffff..

**Pull request merged**